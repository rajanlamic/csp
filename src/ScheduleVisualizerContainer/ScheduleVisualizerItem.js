import React from 'react';
import { Grid } from 'semantic-ui-react';

export const ScheduleVisualizerItem = ({item}) => {
    return (<Grid.Row>
          <Grid.Column width={4}>{item.startTime}</Grid.Column>
          <Grid.Column width={4}>{item.endTime}</Grid.Column>
          <Grid.Column width={8}>{item.titleName}</Grid.Column>
        </Grid.Row>)
}
