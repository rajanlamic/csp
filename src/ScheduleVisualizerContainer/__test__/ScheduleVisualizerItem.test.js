import * as React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import { ScheduleVisualizerItem } from '../ScheduleVisualizerItem';

describe('ScheduleVisualizerItem', () => {
  let component, props;
  
  beforeEach(() => {
    props = { item: {
      startTime: "06:00",
      endTime: "07:00",
      titleName: "The Amazing Spiderman"
    }}
    component = shallow(<ScheduleVisualizerItem {...props} />);
  })
  afterEach(() => {
    component = null;
    props = null;
  })
  
  it('should render ScheduleVisualizerItem component properly using snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  
})
