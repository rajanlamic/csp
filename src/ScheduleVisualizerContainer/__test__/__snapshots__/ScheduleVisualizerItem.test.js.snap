// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleVisualizerItem should render ScheduleVisualizerItem component properly using snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    item={
        Object {
            "endTime": "07:00",
            "startTime": "06:00",
            "titleName": "The Amazing Spiderman",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <GridColumn
          width={4}
>
          06:00
</GridColumn>,
        <GridColumn
          width={4}
>
          07:00
</GridColumn>,
        <GridColumn
          width={8}
>
          The Amazing Spiderman
</GridColumn>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "06:00",
          "width": 4,
        },
        "ref": null,
        "rendered": "06:00",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "07:00",
          "width": 4,
        },
        "ref": null,
        "rendered": "07:00",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "The Amazing Spiderman",
          "width": 8,
        },
        "ref": null,
        "rendered": "The Amazing Spiderman",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GridColumn
            width={4}
>
            06:00
</GridColumn>,
          <GridColumn
            width={4}
>
            07:00
</GridColumn>,
          <GridColumn
            width={8}
>
            The Amazing Spiderman
</GridColumn>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "06:00",
            "width": 4,
          },
          "ref": null,
          "rendered": "06:00",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "07:00",
            "width": 4,
          },
          "ref": null,
          "rendered": "07:00",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "The Amazing Spiderman",
            "width": 8,
          },
          "ref": null,
          "rendered": "The Amazing Spiderman",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
